crates/wasm-host/src/plugin.rs

```diff
@@ -783,13 +792,20 @@
         request_id: u64,
         name: String,
         value: Vec<u8>,
-    ) -> Result<(), wasmtime::Error> {
-        let mut outbound_requests = self.outbound_http.lock().unwrap();
-        // remove/insert to avoid move issues
-        let mut builder = outbound_requests.remove(&request_id).unwrap();
-        builder = builder.header(name, value);
-        outbound_requests.insert(request_id, builder);
-        Ok(())
+    ) -> Result<Result<(), bulwark_host::HttpError>, wasmtime::Error> {
+        Ok(
+            // Inner function to permit ? operator
+            || -> Result<(), bulwark_host::HttpError> {
+                let mut outbound_requests = self.outbound_http.lock().unwrap();
```

Comment:
Right now the unwraps on mutexes are intentional since the mutex would be poisoned. It might be fine to ignore the poisoned mutex given that mutexes here will be disposed of at the end of the request lifecycle, but for the time being, a panic is probably safer at this stage of development and also probably more useful.

Reply:
Also, I think the only way for these locks to become poisoned in the first place is for another function in this part of the code to panic while holding the same mutex locked. So by getting rid of all the other panics, I think it becomes much harder to panic here. I think there's a few remaining panic scenarios that don't involve unwraps though. Either way, panicking in a plugin doesn't terminate the program, just the plugin.

Reply:
I'll update these to `expect` for clarity.

Reply:
Also added a quick Semgrep rule for the future:

```yaml
rules:
- id: unwrap-mutex
  pattern: (...) .lock().unwrap()
  message: Prefer expect when handling a Mutex lock Result
  languages: [rust]
  fix-regex:
    regex: 'unwrap\w*\(\)'
    replacement: 'expect("poisoned mutex")'
  severity: WARNING
```